revus/app/git_operations.py

```diff
@@ -2,51 +2,103 @@
 
 import os
 import sys
+from typing import Dict, List, Union
 from git import Repo, InvalidGitRepositoryError, GitCommandError
 from .config import get_config
 from .logger import log_error
 from .cli import parse_cli_args
 
 
-def get_file_changes():
+def _get_changed_files(repo: Repo) -> Dict[str, List[str]]:
+    """Retrieve the lists of changed files in the Git repository."""
+    try:
+        staged_files = [item.a_path for item in repo.index.diff("HEAD", staged=True)]
+        unstaged_files = repo.git.diff(name_only=True).splitlines()
+        new_files = repo.untracked_files
+        return {
+            "staged": staged_files,
+            "unstaged": unstaged_files,
+            "new": new_files,
+        }
+    except (InvalidGitRepositoryError, GitCommandError) as e:
+        log_error(f"Error working with Git repository: {e}")
```

Comment:
The method could be made a little more readable, in my opinion. It wouldn't hurt to add a couple of comments, too, to make it a little clearer what's going on.

Check separately if it works like this:
```
unstaged_diff = repo.index.diff(None)
changed_files["unstaged"] = [item.a_path for item in unstaged_diff]
```

I think it would be more universal, but I'm not sure it works correctly. If it doesn't, you can return it to how it was.

```python
def _get_changed_files(repo: Repo) -> Dict[str, List[str]]:
    """Retrieves lists of changed files in the Git repository."""

    changed_files = {
        "staged": [],
        "unstaged": [],
        "new": [],
    }

    try:
        # Retrieve staged (indexed) files
        staged_diff = repo.index.diff("HEAD")
        changed_files["staged"] = [item.a_path for item in staged_diff]

        # Retrieve unstaged (modified but not staged) files
        unstaged_diff = repo.index.diff(None)
        changed_files["unstaged"] = [item.a_path for item in unstaged_diff]

        # Retrieve untracked (new) files
        changed_files["new"] = repo.untracked_files

    except (InvalidGitRepositoryError, GitCommandError) as e:
        log_error(f"Error working with Git repository: {e}")

    return changed_files
```